### PR TITLE
Various fixes and improvements

### PR DESCRIPTION
**Pull request by qwerty287**
* locally serve (don't use external servers):
    * jQuery
    * Font Awesome
    * Other fonts **please note that I replaced the current font because it is non-free and can't be served locally without paying for it**, but the font I used (Larke Neue) looks similar to the old ones.
* JS libraries:
    * updated PrismJS, also include more definitions for better highlighting
    * remove require.js, didn't work with new Prism version
* docs content:
    * fix some spelling mistakes
    * fix and improve some styles
    * no longer a default language (PHP), every code block should set this on it's own because there shouldn't be highlighting on inline code
    * CSS cleanups to remove unused selectors and rules (this is responsible for most of the diff)
* view favicon on index and support pages

Overall, this:
* improves privacy since no connection to external servers are initiated
* improves usability of code blocks and its syntax highlighting
* speed up loading since CSS files are smaller and files not longer necessary were removed

If you don't agree with a change this introduces, I'll revert them.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -10,7 +9,7 @@ To make this a cinch, Lychee utilizes the [DotEnv][1] PHP library by Vance Lucas
...
 
-If you are developing with a team, you may wish to continue including a .env.example file with your application. By putting placeholder values in the example configuration file, other developers on your team can clearly see which environment variables are needed to run your application. You may also create a .env.testing file. This file will override the .env file when running PHPUnit tests or executing Artisan commands with the --env=testing option.
+If you are developing with a team, you may wish to continue including an `.env.example` file with your application. By putting placeholder values in the example configuration file, other developers on your team can clearly see which environment variables are needed to run your application. You may also create an `.env.testing` file. This file will override the `.env` file when running PHPUnit tests or executing Artisan commands with the `--env=testing` option.
```
Comment (d7415) [inline docs/configuration.md:12]: I agree with the ``s. I'm curious about the group opinion on "a" vs "an" with a leading dot. I read them as "dot env dot example", so would use "a".

File: docs/upgrade.md [hunk 2]
```diff
@@ -53,7 +53,7 @@ Modify your `/etc/apache2/apache2.conf` to allow `.htaccess` to set up the rewri
 ```
 
 Modify or create `example.com.conf` in `/etc/apache2/sites-available/` to point out the served directory:
-```
+```nginx
```
Comment (d7415) [inline docs/upgrade.md:56]:
```suggestion
```apacheconf
```

**Comment by nagmat84** [comment]
I wonder what we should do about `docs/api.md`. Basically, the whole file is outdated. I have been looking for some tool which would allow us to generate an API description from source code annotations on the controller methods, but I haven't found one. I would have even taken the trouble to comment all controller methods, if there had been such a tool. But I didn't feel like writing a documentation which would have been completely detached from the code, because it will soon become inconsistent again.

**Comment by ildyria** [comment]
About swagger, we do use it at work, but only from yaml.
In our case I believe the followings are more interesting:
- actually swagger php _parser_: https://zircote.github.io/swagger-php/
- example: https://github.com/zircote/swagger-php/blob/master/Examples/petstore.swagger.io/Controllers/StoreController.php

From the look of it, it is mostly generating a nice page from parsing documentation.

**Comment by nagmat84** [comment]
During my search for a tool, I also stumbled across Swagger, but from what I understood, the tool only provides the wrong direction. One can write an API specification using a syntax defined by Swagger and then create a nicely readable API documentation in some other format out of it (e.g. HTML, PDF, etc.) and also generate a code skeleton (e.g. in PHP).

But that is not what I was looking for because our PHP code is already there. Even if one would ignore that part of Swagger, then everything that would be left is that one had to document the API manually (in the Swagger syntax) which would not provide any benefits over maintaining the MD file manually.

> Whatever we do, it requires a lot of work [...] and is out of scope for this PR or even repo.

Agreed. I only mentioned this issue, because the file `api.md`  was part of the merge diff.

> Maybe we could delete the whole file since it's completely outdated and replace it at some time with an auto-generated one?

I doubt that we will ever replace the file by an auto-generated one, because there simply seems no tool for it. If we are planning of keeping something like that, we will have to write it manually at some time. IMHO, there are two viable options: 

a) we delete the file

b) In case, we want to have that file back at some time, we might keep it as a reminder but then I would add a big, fad warning to the beginning of the file that it is completely outdated and that a reader should look at `routes.php` + friends, `App\Http\Requests` and `App\Http\Exceptions` for the ultimate truth.

**Comment by ildyria** [comment]
@qwerty287  please check the changes and merge if you are happy with it. :)

**Comment by nagmat84** [comment]
> > * add @querty287 to contributor list because they deserve it.
> 
> If you need a pronoun for me again, you can use `he`. :)

The "pluralis majestatis" suits you well. :grinning:

**Pull request merged**